src/test/scala/chiselTests/ModuleChoiceSpec.scala

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests
+
+import chisel3._
+import chisel3.parameterization.{Case, OptionGroup}
+import chiselTests.{ChiselFlatSpec, MatchesAndOmits, Utils}
+import _root_.circt.stage.ChiselStage
+
+object Platform extends OptionGroup {
+  object FPGA extends Case
+  object ASIC extends Case
+}
+
+class TargetIO extends Bundle {
```

Comment:
what happens if TargetIO takes a parameter and the different targets differ in their (new TargetIO(6) vs new TargetIO(8))? Mostly just curious where this error would be detected and what the error message looks like

Reply:
`firtool` definitely checks for this. Should we also verify the structure of the IO bundles in Chisel?

Reply:
Added a check to ensure that the bundles are type equivalent.